component/component/main.jsonnet

```diff
@@ -94,27 +94,30 @@ local cronjob(name, subcommand, schedule) = {
   },
 };
 
-local objectStorageSchedule =
-  if std.objectHas(params, 'schedule') then
-    std.trace(
-      'Parameter `schedule` is deprecated,'
-      + ' please use parameter `objectStorage.schedule` instead',
-      params.schedule
-    )
-  else
-    params.objectStorage.schedule;
+(if params.exoscale.enabled then {
+  local secrets = params.secrets['exoscale'],
+  assert secrets != null : 'secrets must be set.',
+  assert secrets.credentials != null : 'secrets.credentials must be set.',
+  assert secrets.credentials.stringData != null : 'secrets.credentials.stringData must be set.',
+  assert secrets.credentials.stringData.EXOSCALE_API_KEY != null : 'secrets.credentials.stringData.EXOSCALE_API_KEY must be set.',
+  assert secrets.credentials.stringData.EXOSCALE_API_SECRET != null : 'secrets.credentials.stringData.EXOSCALE_API_SECRET must be set.',
+  assert secrets.credentials.stringData.K8S_SERVER_URL != null : 'secrets.credentials.stringData.K8S_SERVER_URL must be set.',
+  assert secrets.credentials.stringData.K8S_TOKEN != null : 'secrets.credentials.stringData.K8S_TOKEN must be set.',
 
-{
-  assert params.secrets != null : 'secrets must be set.',
-  assert params.secrets.credentials != null : 'secrets.credentials must be set.',
-  assert params.secrets.credentials.stringData != null : 'secrets.credentials.stringData must be set.',
-  assert params.secrets.credentials.stringData.EXOSCALE_API_KEY != null : 'secrets.credentials.stringData.EXOSCALE_API_KEY must be set.',
-  assert params.secrets.credentials.stringData.EXOSCALE_API_SECRET != null : 'secrets.credentials.stringData.EXOSCALE_API_SECRET must be set.',
-  assert params.secrets.credentials.stringData.K8S_SERVER_URL != null : 'secrets.credentials.stringData.K8S_SERVER_URL must be set.',
-  assert params.secrets.credentials.stringData.K8S_TOKEN != null : 'secrets.credentials.stringData.K8S_TOKEN must be set.',
+  secrets: std.filter(function(it) it != null, secret('exoscale')),
+  objectStorageCronjob: cronjob(alias + '-objectstorage', 'exoscale objectstorage', params.exoscale.objectStorage.schedule),
+  [if params.exoscale.dbaas.enabled then 'dbaasCronjob']: cronjob(alias + '-dbaas', 'exoscale dbaas', params.exoscale.dbaas.schedule),
+} else {})
++
+(if params.cloudscale.enabled then {
+  local secrets = params.secrets['cloudscale'],
```

Comment:
Nitpick: Maybe a check that only cloudscale or exoscale is enabled might be helpful. As far as I can tell currently if both are enabled for an instance, weird things will happen

Reply:
if there are known issues, they have to be resolved asap and preferably in this PR. I would expected the component from master branch to work in any configuration allowed by the component itself.

Reply:
Also I'm not sure if we want to allow the installation of the component when neither exoscale nor cloudscale is installed.

Reply:
> just FTR in case you oversaw that: target branch is still `refactored-collectors` - it doesn't go into master just yet.

yeah I missed that :)